src/General/FileInfo.hs

```diff
@@ -0,0 +1,258 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving, DeriveDataTypeable, CPP, DeriveGeneric #-}
```

Comment:
Can we get a comment saying where it got copied from and at what date/SHA1, in case we ever need to sync future changes or analyse bugs between the two.